x-pack/plugins/ml/public/settings/calendars/edit/new_calendar.js

```diff
@@ -103,17 +103,35 @@ export class NewCalendar extends Component {
     }
   }
 
-  onCreate = async () => {
-    const calendar = this.setUpCalendarForApi();
-    this.setState({ saving: true });
+  isDuplicateId = () => {
+    const { calendars, formCalendarId } = this.state;
 
-    try {
-      await ml.addCalendar(calendar);
-      window.location = `${chrome.getBasePath()}/app/ml#/settings/calendars_list`;
-    } catch (error) {
-      console.log('Error saving calendar', error);
-      this.setState({ saving: false });
-      toastNotifications.addDanger(`An error occurred creating calendar ${calendar.calendarId}`);
+    for (let i = 0; i < calendars.length; i++) {
+      if (calendars[i].calendar_id === formCalendarId) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  onCreate = async () => {
+    const { formCalendarId } = this.state;
+
+    if (this.isDuplicateId()) {
+      toastNotifications.addWarning(`Cannot create calendar with id [${formCalendarId}] as it already exists.`);
```

Comment:
Looking at the eui guidelines, I think we should use `addDanger` here as the form has an 'error' in that the ID is invalid,. Would also make it consistent with the Filter Lists 'duplicate ID' toast.